x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/new_terms/create_new_terms_alert_type.ts

```diff
@@ -188,88 +185,87 @@ export const createNewTermsAlertType = (
         result.searchAfterTimes.push(searchDuration);
         result.errors.push(...searchErrors);
 
-        afterKey = searchResultWithAggs.aggregations.new_terms.after_key;
-
         // If the aggregation returns no after_key it signals that we've paged through all results
         // and the current page is empty so we can immediately break.
-        if (afterKey == null) {
+        if (searchResultWithAggs.aggregations.new_terms.after_key == null) {
           break;
         }
         const bucketsForField = searchResultWithAggs.aggregations.new_terms.buckets;
-        const includeValues = transformBucketsToValues(params.newTermsFields, bucketsForField);
-        const newTermsRuntimeMappings = getNewTermsRuntimeMappings(
-          params.newTermsFields,
-          bucketsForField
-        );
 
-        // PHASE 2: Take the page of results from Phase 1 and determine if each term exists in the history window.
-        // The aggregation filters out buckets for terms that exist prior to `tuple.from`, so the buckets in the
-        // response correspond to each new term.
-        const {
-          searchResult: pageSearchResult,
-          searchDuration: pageSearchDuration,
-          searchErrors: pageSearchErrors,
-        } = await singleSearchAfter({
-          aggregations: buildNewTermsAgg({
-            newValueWindowStart: tuple.from,
-            timestampField: aggregatableTimestampField,
-            field: getAggregationField(params.newTermsFields),
-            include: includeValues,
-          }),
-          runtimeMappings: {
-            ...runtimeMappings,
-            ...newTermsRuntimeMappings,
-          },
-          searchAfterSortIds: undefined,
-          index: inputIndex,
-          // For Phase 2, we expand the time range to aggregate over the history window
-          // in addition to the rule interval
-          from: parsedHistoryWindowSize.toISOString(),
-          to: tuple.to.toISOString(),
-          services,
-          ruleExecutionLogger,
-          filter: esFilter,
-          pageSize: 0,
-          primaryTimestamp,
-          secondaryTimestamp,
-        });
-        result.searchAfterTimes.push(pageSearchDuration);
-        result.errors.push(...pageSearchErrors);
-
-        logger.debug(`Time spent on phase 2 terms agg: ${pageSearchDuration}`);
+        const createAlertsHook: CreateAlertsHook = async (aggResult) => {
```

Comment:
I created this function in closure, so it can use a lot context values within executor without a need to create extensive interface and passing a dozen of properties as arguments there. It is also heavily depend on new terms logic, i.e. creating `eventsAndTerms`, wrapping new terms alerts(which use new terms alerts interface), which makes it even more difficult to parameterised it for other rule types. So, I would like to leave it, enclosed within executor, to make maintenance burden as simple as it can be.